.github/workflows/ci-lint-test.yml

```diff
@@ -0,0 +1,38 @@
+name: "CI: Lint & Test"
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: "[INIT] Checkout project"
+        uses: actions/checkout@v4
+      - name: "[EXEC] Lint"
+        uses: chartboost/ruff-action@v1
+  test:
+    name: Test
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash
+    strategy:
+      matrix:
+        python: ["3.8", "3.9", "3.10", "3.11", "3.12", "pypy3.9", "pypy3.10"]
+        os: [ubuntu-latest, macos-12, windows-latest]
```

Comment:
Use `macos-latest` here

Reply:
last time i tried this, macos latest uses a different architecture (ARM), which didn't have older python versions at the time. unless this has changed, that will not be possible.

Reply:
You are right, seems that changed recently. macos-latest used to point to macos-13, now it points to 14 which uses M1 cpu's.

https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories

So in theory you can use `macos-13`, instead of 12

Reply:
that's especially important now that macos-12 is EOL. macos-13 has another year of support, so i'll resolve this later.